R/module_filter_data.R

```diff
@@ -62,9 +62,12 @@ srv_filter_data <- function(id, datasets, active_datanames, data_rv, is_active)
     )
   )
   filtered_code <- .get_filter_expr(datasets = datasets, datanames = datanames)
+  # Can below 3 lines be substituted with
   filtered_teal_data <- .append_evaluated_code(data, filtered_code)
   filtered_datasets <- sapply(datanames, function(x) datasets$get_data(x, filtered = TRUE), simplify = FALSE)
   filtered_teal_data <- .append_modified_data(filtered_teal_data, filtered_datasets)
+  # this line
+  # filtered_teal_data <- eval_code(data, filtered_code)
```

Comment:
It would be the same

Given this doesn't call `datasets$get_data(x, filtered = TRUE)` the reactive that actual filters the data in `teal.slice` is never called (hence repeated expression are not "run" twice) -- Or I'm misremembering?

I'm still worried about repeated code being executed in the filter panel :sweat_smile:

Reply:
`eval_code` should be removed. Filter-code is added to the qenv when triggered. No double evaluation

Reply:
Right. Removing it.